fsharp-backend/src/LibExecution/RuntimeTypes.fs

```diff
@@ -841,8 +849,17 @@ and FnImpl =
   | UserFunction of tlid * Expr
   | PackageFunction of tlid * Expr
 
+
+// CLEANUP consider renaming to `ExecutionType`, `EvaluationMode`, etc.
+/// Represents the context in which we're evaluating some code
 and RealOrPreview =
+  /// We are evaluating an expression normally, and it is OK to affect state.
   | Real
+
+  /// We are previewing the evaluation of some expression,
```

Comment:
"within a canvas" isn't a precise definition here - all execution happens within a canvas. Perhaps something like "in the editor". Previews only happen in Wasm, we don't do it any other place.